tests/std/tests/P0896R4_views_transform/test.cpp

```diff
@@ -509,16 +509,8 @@ struct iterator_instantiator {
             assert(ranges::iter_move(i0) == add8(mutable_ints[0])); // NB: moving from int leaves it unchanged
             STATIC_ASSERT(NOEXCEPT_IDL0(ranges::iter_move(i0)));
 
-            if constexpr (forward_iterator<Iter>) {
-                auto i1 = ranges::next(i0);
-                ranges::iter_swap(i0, i1);
-                assert(mutable_ints[0] == 1);
-                assert(mutable_ints[1] == 0);
-                ranges::iter_swap(i1, i0);
-                assert(mutable_ints[0] == 0);
-                assert(mutable_ints[1] == 1);
-                STATIC_ASSERT(NOEXCEPT_IDL0(ranges::iter_swap(i0, i1)));
-            }
+            // LWG-3520 forbids iter_swap for transform_view::iterator

```

Comment:
This comment is misleading. Calling `iter_swap` on `transform_view::iterator` is not forbidden, it's just impossible when the result of transformation is not a lvalue reference. We should also add test coverage for that.